cloud-controller-manager/do/kubernetes_lb_validator_test.go

```diff
@@ -0,0 +1,283 @@
+					},
+					Spec: corev1.ServiceSpec{
+						Type: corev1.ServiceTypeLoadBalancer,
+					},
+				},
+				nil,
+			)},
+			expectedAllowed:    false,
+			resp:               newFakeUnprocessableResponse(),
+			err:                newFakeUnprocessableErrorResponse(),
+			expectedStatusCode: int32(http.StatusForbidden),
+		},
+		{
+			name: "Allow Update happy path",
+			req: admission.Request{AdmissionRequest: fakeAdmissionRequest(
+				fakeService("test2"), fakeService("old-service"))},
+			expectedAllowed:    true,
+			expectedStatusCode: int32(http.StatusOK),
+		},
+		{
+			name: "Deny Update invalid configuration",
+			req: admission.Request{AdmissionRequest: fakeAdmissionRequest(
+				fakeService("test2"), fakeService("old-service"))},
+			expectedAllowed:    false,
+			resp:               newFakeUnprocessableResponse(),
+			err:                newFakeUnprocessableErrorResponse(),
+			expectedStatusCode: int32(http.StatusForbidden),
+		},
+		{
+			name: "Deny Update validation error",
+			req: admission.Request{AdmissionRequest: fakeAdmissionRequest(
+				fakeService("test2"), fakeService("old-service"))},
+			expectedAllowed:    false,
+			expectedMessage:    "failed to validate lb update, could not get validation response",
+			resp:               newFakeNotFoundResponse(),
+			err:                newFakeNotFoundErrorResponse(),
+			expectedStatusCode: int32(newFakeNotFoundErrorResponse().Response.StatusCode),
+		},
+	}
+
+	for _, test := range testcases {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			// setup client
+			gClient := godo.NewFromToken("")
+			gClient.LoadBalancers = &fakeLBService{
+				createFn: func(context.Context, *godo.LoadBalancerRequest) (*godo.LoadBalancer, *godo.Response, error) {
+					return &godo.LoadBalancer{ID: "2", Name: "two"}, test.resp, test.err
+				},
+				updateFn: func(context.Context, string, *godo.LoadBalancerRequest) (*godo.LoadBalancer, *godo.Response, error) {
+					return &godo.LoadBalancer{ID: "2", Name: "two"}, test.resp, test.err
+				},
+			}
+			gClient.Regions = &fakeRegionsService{
+				listFn: func(context.Context, *godo.ListOptions) ([]godo.Region, *godo.Response, error) {
+					return []godo.Region{{Name: "nyc3", Slug: "nyc3"}}, newFakeOKResponse(), nil
+				}}
+
+			decoder, err := admission.NewDecoder(scheme)
+			if err != nil {
+				t.Fatalf("failed to initialize decoder %s", err)
+			}
+
+			var logOpts []zap.Opts
+			ll := zap.New(logOpts...).WithName("webhook-validation-server")
+			ctrlruntimelog.SetLogger(ll)
+
+			validator := &KubernetesLBServiceValidator{
+				Log:     ll,
+				decoder: decoder,
+				GClient: gClient,
+			}
+
+			res := validator.Handle(context.TODO(), test.req)
+			if res.Allowed != test.expectedAllowed {
+				t.Fatalf("got allowed %v, want %v", res.Allowed, test.expectedAllowed)
+			}
+			if res.Result.Code != test.expectedStatusCode {
+				t.Fatalf("got allowed %v, want %v", res.Result.Code, test.expectedStatusCode)
+			}
```

Comment:
Here is where we should use `expectedMessage` as well based on the response message that I linked to in another comment.